R/driver-db2.R

```diff
@@ -29,3 +30,44 @@ setMethod("sqlCreateTable", "DB2/AIX64",
     ))
   }
 )
+
+
+#' @rdname DBI-methods
+# In addition to the `ODBC` API call to `SQLTables`, also
+# attempt to query SYSIBMADM.ADMINTEMPTABLES for any tables
+# that match the arguments.
+#' @usage NULL
+setMethod("odbcConnectionTables", c("DB2/AIX64", "character"),
+  function(conn,
+           name,
+           catalog_name = NULL,
+           schema_name = NULL,
+           table_type = NULL,
+           exact = FALSE) {
+
+    res <- callNextMethod()
+    dfTempTables <- NULL
+    if ( (is.null(schema_name) || schema_name == "SESSION") &&
```

Comment:
I almost wonder if it's worth adding a `odbConnectionTempTables()` generic that we could call from `odbcConnectionTables()`. Not for this PR, but might be interesting to consider in the future.

Reply:
Yep  - we've needed to do this a few times now.  Probably a good idea.